### PR TITLE
fix i3bar relative coordinates when scaling is used

### PR DESCRIPTION
**Pull request by hvenev**
The first commit fixes the mistakes I made in 24e8ba048aef4751c6fa1d5982ee634f921e6cf6. The second commit is the one that actually increases the precision of the reported values. Pixel coordinates are sent, not logical coordinates. Do you think this makes sense, or should logical coordinates be sent as floating-point values?

I haven't tested multi-monitor setups.

### CODE REVIEW

**Comment by emersion** [comment]
>Pixel coordinates are sent, not logical coordinates. Do you think this makes sense, or should logical coordinates be sent as floating-point values?

I think logical coords would make more sense. Current bar programs don't support the "scale" parameter.

Do current clients support floating point coords?

**Comment by hvenev** [comment]
> I think logical coords would make more sense. Current bar programs don't support the "scale" parameter.

You are right, logical coordinates are better.

> Do current bar programs support floating point coords?

Most of the ones written in python probably will, but I have no idea what JSON implementations in statically typed programming languages do.

We could have the bar program announce that it supports floating-point coordinates, like we do with click events.

File: swaybar/status_line.c [hunk 1]
```diff
@@ -85,6 +85,13 @@ bool status_handle_readable(struct status_line *status) {
 				}
 			}
 
+			json_object *float_coords;
+			if (json_object_object_get_ex(header, "float_coords", &float_coords)
```
Comment (emersion) [inline swaybar/status_line.c]: Maybe name this `float_event_coords`, to make it clear it's about events.

File: swaybar/input.c [hunk 2]
```diff
@@ -229,19 +231,8 @@ static void wl_pointer_axis(void *data, struct wl_pointer *wl_pointer,
...
-		if (x >= hotspot->x
-				&& y >= hotspot->y
-				&& x < hotspot->x + hotspot->width
-				&& y < hotspot->y + hotspot->height) {
-			if (HOTSPOT_IGNORE == hotspot->callback(output, hotspot, x, y,
-					button, hotspot->data)) {
-				return;
-			}
-		}
+	if(process_hotspots(output, pointer->x, pointer->y, button)) {
```
Comment (emersion) [inline swaybar/input.c]: Style: missing space after `if`

File: swaybar/status_line.c (hunk 1 above)
Comment (hvenev) [reply swaybar/status_line.c]: Done.

File: swaybar/input.c (hunk 2 above)
Comment (hvenev) [reply swaybar/input.c]: Fixed.

**Comment by emersion** [comment]
Thanks!

**Pull request merged**